cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,234 @@
-from cfme.fixtures import pytest_selenium as sel
-from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from navmazing import NavigateToSibling, NavigateToAttribute, NavigateToObject
+from widgetastic.exceptions import NoSuchElementException
+from cfme.exceptions import RoleNotFound
+from widgetastic.widget import View
+from widgetastic_patternfly import BootstrapNav, Button, Dropdown, FlashMessages
+from widgetastic_manageiq import (
+    Accordion, BootstrapSelect, BreadCrumb, ItemsToolBarViewSelector, PaginationPane,
+    SummaryTable, Table, Text, BootstrapTreeview)
+
+from cfme.base.ui import BaseLoggedInPage
 from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import (CFMENavigateStep, navigate_to,
-                                                navigator)
-from cfme.web_ui import (listaccordion as list_acc, match_location, Quadicon,
-                         Region, toolbar as tb)
-from utils import version
-from utils.pretty import Pretty
-
-
-details_page = Region(infoblock_type='detail')
-
-cfg_btn = partial(tb.select, 'Configuration')
-pol_btn = partial(tb.select, 'Policy')
-match_page = partial(match_location, controller='ems_cluster')
-
-
-class DeploymentRoles(Pretty, Navigatable):
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from utils.log import logger
+from utils.wait import wait_for, TimedOutError
+from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from cfme.common.provider_views import InfraProviderDetailsView
+
+
+class DeploymentRoleToolbar(View):
+    """The toolbar on the Deployment Role page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class DeploymentRoleDetailsToolbar(View):
+    """The toolbar on the Deployment Role details page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    monitoring = Dropdown('Monitoring')
+    download = Button('Download summary in PDF format')
+
+
+class DeploymentRoleComparisonToolbar(View):
+    """ The toolbar on Comparison Page of roles"""
+    all_attributes = Button(title="All attributes")
+    different_values_attributes = Button(title="Attributes with different values")
+    same_values_attributes = Button(title="Attributes with same values")
+
+    details_mode = Button(title="Details Mode")
+    exists_mode = Button(title="Exists Mode")
+    download = Dropdown('Download')
+
+
+class DeploymentRoleDetailsAccordion(View):
+    """The accordion on the Deployment Role details page"""
+    @View.nested
+    class properties(Accordion):        # noqa
+        nav = BootstrapNav('//div[@id="ems_prop"]//ul')
+
+    @View.nested
+    class relationships(Accordion):     # noqa
+        nav = BootstrapNav('//div[@id="ems_rel"]//ul')
+
+
+class DeploymentRoleEntities(View):
```

Comment:
**REQUIRED**
It should inherit `BaseEntitiesView`. Usin entities you don't use resetter in a navigation step. Please look at [cfme.common.provider_views](https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/provider_views.py) and [cfme.common.host_views](https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/host_views.py) as a reference.

Reply:
@quarckster done

Reply:
DeploymentRoleEntitiesView inherited with BaseEntitiesView and added JS and NonJS class.